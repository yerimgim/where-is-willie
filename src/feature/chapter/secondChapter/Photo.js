import React from "react";

import PropTypes from "prop-types";
import styled from "styled-components";

const Photo = ({ setIsOpen, isOpen }) => {
  const onClick = () => {
    setIsOpen(!isOpen);
  };

  return (
    <Entry>
      <div className="photo-container" onClick={onClick}>
        <img src="assets/people.png" />
      </div>
    </Entry>
  );
};

const Entry = styled.div`
  position: absolute;
  top: 75%;
  left: 33%;
  width: 10%;
  height: 10%;

  .photo-container {
    display: flex;
    justify-content: center;
  }

  img {
    width: 25%;
    opacity: 0.6;
    transform: skew(330deg, 20deg);

    &:hover {
      cursor: pointer;
      filter: drop-shadow(0.3rem 0.3rem 0.8rem var(--yellow-color));
    }
  }
`;

Photo.propTypes = {
  setIsOpen: PropTypes.func,
  isOpen: PropTypes.bool,
};

export default Photo;
