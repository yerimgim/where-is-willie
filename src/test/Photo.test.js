import React from "react";

import { cleanup, render, screen } from "@testing-library/react";

import Photo from "../feature/chapter/secondChapter/Photo";

afterEach(cleanup);

describe("<Photo />", () => {
  test("<Photo /> - onClick", () => {
    const { container } = render(<Photo variant="default" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("<Photo /> - img alt property ", () => {
    const { getByAltText } = render(<Photo />);

    getByAltText("사진 모음 이미지");
    expect(screen.getByAltText("사진 모음 이미지")).toBeTruthy();
  });

  test("<Photo /> - img src", () => {
    render(<Photo />);
    const people = screen.getByRole("img");

    expect(people).toHaveAttribute("src", "assets/people.png");
  });
});
