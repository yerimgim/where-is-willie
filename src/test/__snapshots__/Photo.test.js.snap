// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Photo /> <Photo /> - onClick 1`] = `
<div
  class="sc-bdvvtL gFPPuV"
>
  <div
    class="photo-container"
  >
    <img
      alt="사진 모음 이미지"
      src="assets/people.png"
    />
  </div>
</div>
`;
